File: LeetCode/isomorphic_strings.rb
# Given two strings s and t, determine if they are isomorphic.
#
# two strings are isomorphic if the characters in s can be replaced to get t.

Def isomorphic?(s, t)
    a_s = s.split("")
    a_t = t.split("")
    a_s.map{|x| a_s.index(x)} == a_t.map{|x| a_t.index(x)}
end

puts isomorphic?("egg", "add")
puts isomorphic?("foo", "bar")
puts isomorphic?("paper", "title")

File: LeetCode/isomorphic_strings.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program {
    static bool Isomorphic(String s, String t) {
        return s.Select(x => s.IndexOf(x)).SequenceEqual(t.Select(x => t.IndexOf(x)));
    }

    static void Main() {
        var input = new List<Tuple<String, String>>() {
            new Tuple<String, String>("egg", "add"),
            new Tuple<String, String>("foo", "bar"),
            new Tuple<String, String>("paper", "title")
        };

        Console.WriteLine(String.Join(",", input.Select(x => String.Format("{0} - {1}", x, Isomorphic(x.Item1, x.Item2)))));
    }
}
